core/services/relay/evm/mercury/v1/data_source.go

```diff
@@ -292,39 +291,30 @@ func (ds *datasource) executeRun(ctx context.Context) (*pipeline.Run, pipeline.T
 }
 
 func (ds *datasource) setLatestBlocks(ctx context.Context, obs *relaymercuryv1.Observation) {
-	latestBlocks := ds.getLatestBlocks(ctx, nBlocksObservation)
+	latestBlocks, err := ds.chainReader.LatestHeads(ctx, nBlocksObservation)
```

Comment:
Why don't we handle this error right away?

Reply:
The error is only used in the observation, what's your suggestion?

Reply:
I'm not sure. It is unusual to interact with the value before checking the error. Will it be confusing that the logs for successful empty response look the same as a failed call? That seems significant.

Reply:
the interaction is the length check. I agree that it looks unusual, but the error not handled here but passed with the observation. @samsondav wdyt ?

Reply:
Why exclude the error from the log line and drop it all together though? It won't be clear when there was a successful empty response vs an actual failure.

Reply:
good point, will log.